src/D2L.Security.OAuth2/Keys/Default/JsonWebKeySet.cs

```diff
@@ -17,13 +18,23 @@ internal sealed class JsonWebKeySet {
 			Source = src ?? throw new ArgumentNullException( nameof( src ) );
 
 			try {
-				var data = JsonSerializer.Deserialize<Dictionary<string, List<object>>>( json );
+				var data = JsonSerializer.Deserialize<Dictionary<string, object>>( json );
 
 				if( !data.ContainsKey( "keys" ) ) {
 					throw new JsonWebKeyParseException( "invalid json web key set: missing keys array" );
 				}
 
-				List<object> keyObjects = data["keys"];
+				#if NET6_0
+				System.Text.Json.JsonElement keysElement = ( System.Text.Json.JsonElement )data["keys"];
+				if ( keysElement.ValueKind is not System.Text.Json.JsonValueKind.Array ) {
+					throw new JsonWebKeyParseException("invalid json web key set: keys not an array");
+				}
+				List<System.Text.Json.JsonElement> keyObjects = keysElement.EnumerateArray().ToList();

```

Comment:
For .NET 6 we might want to just switch to `JsonDocument` in .NET 6 vs. `JsonSerializer.Deserialize`+the casts/reflection, but I could take that for later, this is fine.